docs: Tyk MDCB 2.11.0 release notes.mdx (#1902)
TykTechnologies/tyk-docs: developer-support/release-notes/mdcb.mdx

#### Deprecations
There are no deprecations in this release.

#### Upgrade instructions
If you are upgrading to 2.11.0, please follow the detailed [upgrade instructions](#upgrading-tyk).

For distributed deployments, deploy MDCB 2.11.0 before upgrading Tyk Pump to 1.15.0 and Tyk Gateway to 5.13.0. MCP analytics support is additive and does not affect existing Pump or Gateway versions until they are also upgraded. See [MCP Gateway upgrade considerations](/developer-support/upgrading#mcp-upgrade-considerations) for the full deployment order.

#### Downloads
- [Docker image v2.11.0](https://hub.docker.com/r/tykio/tyk-mdcb-docker/tags?page=&page_size=&ordering=&name=v2.11.0)
  - ```bash
    docker pull tykio/tyk-mdcb-docker:v2.11.0
    ```
#### Changelog
<a id="Changelog-v2.11.0" data-scroll-offset></a>

##### Added

<AccordionGroup>

<Accordion title='Add support for MCP Proxy analytics'>
MDCB 2.11.0 introduces support for routing MCP analytics to the Control Plane in distributed deployments. MCP analytics records — both raw and pre-aggregated — are routed separately from standard API analytics, keeping MCP traffic isolated in your analytics storage backend.

> No additional MDCB configuration is required. For details, see the [MCP analytics documentation](/ai-management/mcp-gateway/mcp-analytics).

Reviewer: ```suggestion
No additional MDCB configuration is required. For details, see the [MCP analytics documentation](ai-management/mcp-gateway/mcp-observability).
```

Given the time difference between your upgrade and the release of this version, we recommend customers verify the ongoing support of third-party dependencies they install, as their status may have changed since the release.

#### Deprecations
There are no deprecations in this release.

#### Upgrade instructions
If you are upgrading to 2.11.0, please follow the detailed [upgrade instructions](#upgrading-tyk).

#### Downloads
- [Docker image v2.11.0](https://hub.docker.com/r/tykio/tyk-mdcb-docker/tags?page=&page_size=&ordering=&name=v2.11.0)
  - ```bash
    docker pull tykio/tyk-mdcb-docker:v2.11.0
    ```
#### Changelog
<a id="Changelog-v2.11.0" data-scroll-offset></a>

##### Added

<AccordionGroup>

<Accordion title='Add dedicated MCP analytics RPC dispatchers for hybrid deployments'>
MDCB 2.11.0 introduces two new RPC dispatcher methods that give hybrid Data Planes an isolated channel for routing MCP analytics to the Control Plane:

- [PurgeAnalyticsDataMCP](/tyk-multi-data-centre/mdcb-configuration-options#purgeanalyticsdatamcp): routes raw MCP analytics records from the Data Plane to the Control Plane's configured analytics storage backend (MongoDB, PostgreSQL, etc.)

> - [PurgeAnalyticsDataMCPAggregated](/tyk-multi-data-centre/mdcb-configuration-options#purgeanalyticsdatamcpaggregated): routes pre-aggregated MCP analytics records, enabling efficient storage of high-volume MCP traffic without raw record overhead

Reviewer: ```suggestion
- `PurgeAnalyticsDataMCP`: routes raw MCP analytics records from the Data Plane to the Control Plane's configured analytics storage backend (MongoDB, PostgreSQL, etc.)
- `PurgeAnalyticsDataMCPAggregated`: routes pre-aggregated MCP analytics records, enabling efficient storage of high-volume MCP traffic without raw record overhead
```

---

## 2.11 Release Notes

### 2.11.0 Release Notes

#### Release Date 8 May 2026

#### Release Highlights

> MDCB 2.11.0 brings MCP analytics support to hybrid deployments. Data Planes can now route MCP traffic records to the Control Plane through dedicated, isolated channels, keeping MCP analytics separate from standard API analytics and supporting both raw and aggregated record storage.

Reviewer 2: What does this have to do with hybrid? Do you mean "distributed deployments" or "multiple Data Plane deployments"?

Reviewer 2: Ensure you include a link to /ai-management/mcp-gateway/managing-mcp-proxies

Reviewer 3: I think "distributed deployments" is the right choice here; the whole change was about enabling MCP metrics for every kind of tyk deployment mode.

> MDCB 2.11.0 brings MCP analytics support to distributed deployments. Data Planes can now route MCP traffic records to the Control Plane through dedicated, isolated channels, keeping MCP analytics separate from standard API analytics and supporting both raw and aggregated record storage.

Reviewer 2: Too much detail here - we are simply adding support for MCP analytics.

#### Breaking Changes
This release has no breaking changes.

#### Dependencies
<a id="dependencies-2.11.0"></a>

##### 3rd Party Dependencies & Tools
| Third Party Dependency          | Tested Versions | Compatible Versions | Comments |
| :------------------------------- | :--------------- | :------------------- | :-------- |
| [Redis](https://redis.io/download/)    | 6.2.x, 7.x, 7.4.x      | 6.2.x, 7.x, 7.4.x      | |
| [Valkey](https://valkey.io/download/)  | 7.2.x, 8.0.x, 8.1.x    | 7.2.x, 8.0.x, 8.1.x    | |
| [MongoDB](https://www.mongodb.com/try/download/community)  | 5.0.x, 6.0.x, 7.0.x | 4.4.x, 5.0.x, 6.0.x, 7.0.x | |
| [PostgreSQL](https://www.postgresql.org/download/)         | 13.x - 17.x        | 13.x - 17.x            | |

Given the time difference between your upgrade and the release of this version, we recommend customers verify the ongoing support of third-party dependencies they install, as their status may have changed since the release.

#### Deprecations
There are no deprecations in this release.

#### Upgrade instructions
If you are upgrading to 2.11.0, please follow the detailed [upgrade instructions](#upgrading-tyk).

> For hybrid deployments using MCP Gateway, deploy MDCB 2.11.0 before upgrading Tyk Pump to 1.15.0 and Tyk Gateway to 5.13.0. The new MCP analytics RPC handlers are additive and do not affect existing Pump or Gateway versions until they are also upgraded. See [MCP Gateway upgrade considerations](/developer-support/upgrading#mcp-upgrade-considerations) for the full deployment order.

Reviewer 2: Again, why Hybrid?

There is no such thing as MCP Gateway, it's just Tyk Gateway 5.13.0 or later.

No need to mention "RPC handlers" - this is internal to MDCB, I'm not sure why users need to know this.

Just give the clear instructions: upgrade MDCB before upgrading the Data Plane Gateways and, if used, Tyk Pump.

Reviewer 3: I think the RPC stuff was added due to on of our concerns about the upgrade order. The main idea was that it's safe to update MDCB first.

#### Downloads
- [Docker image v2.11.0](https://hub.docker.com/r/tykio/tyk-mdcb-docker/tags?page=&page_size=&ordering=&name=v2.11.0)
  - ```bash
    docker pull tykio/tyk-mdcb-docker:v2.11.0
    ```
#### Changelog
<a id="Changelog-v2.11.0" data-scroll-offset></a>

##### Added

<AccordionGroup>

> <Accordion title='Add dedicated MCP analytics RPC dispatchers for distributed deployments'>

Reviewer 2: This entry should simply cover the fact that we have added support for MCP analytics. We do not need to give details of the internal changes - just the user impact.

> <Accordion title='Add dedicated MCP analytics RPC dispatchers for hybrid deployments'>

Reviewer 2: This is unnecessary detail and again, why "hybrid deployment"?

Just something like:

```suggestion
<Accordion title='"Added support for MCP Proxy analytics'>
```

> MDCB 2.11.0 introduces two new RPC dispatcher methods that give hybrid Data Planes an isolated channel for routing MCP analytics to the Control Plane:

Reviewer 2: We don't talk about "RPC dispatcher methods" in the documentation, this is unnecessary implementation detail. And, again, "hybrid Data Planes" is wrong.

```suggestion
MDCB 2.11.0 introduces support for routing MCP analytics to the Control Plane. There are two new configuration options:
```

> - [PurgeAnalyticsDataMCP](/tyk-multi-data-centre/mdcb-configuration-options#purgeanalyticsdatamcp): routes raw MCP analytics records from the Data Plane to the Control Plane's configured analytics storage backend (MongoDB, PostgreSQL, etc.)

Reviewer 2: Use the correct config name, not internal camelCase name.

Why are these called `purgeanalytics...`?

This doesn't follow existing naming conventions where all the "analytics" controls are within the `analytics` structure (https://tyk.io/docs/tyk-multi-data-centre/mdcb-configuration-options#analytics)

Reviewer 2: N/A as this is all internal

> - [PurgeAnalyticsDataMCPAggregated](/tyk-multi-data-centre/mdcb-configuration-options#purgeanalyticsdatamcpaggregated): routes pre-aggregated MCP analytics records, enabling efficient storage of high-volume MCP traffic without raw record overhead

Reviewer 2: Again, naming... and this one is vague "raw record overhead". What does this mean to the user?

Reviewer 2: N/A this is internal detail

> These dispatchers work alongside the existing `PurgeAnalyticsData` and `PurgeAnalyticsDataAggregated` methods. MCP traffic is routed separately, so MCP analytics are not mixed with standard API analytics in hybrid deployments.

Reviewer 2: This is completely unnecessary detail

> These dispatchers work alongside the existing `PurgeAnalyticsData` and `PurgeAnalyticsDataAggregated` methods. MCP traffic is routed separately, so MCP analytics are not mixed with standard API analytics in distributed deployments.

Reviewer 2: What are these "existing methods"? Is this internal detail? I don't see anything in the config/env var docs for these.

Reviewer 2: N/A this is internal detail to be removed
